collection/Collections.java

```diff
@@ -93,6 +95,27 @@ public static <T> List<T> list(Collection<T> collection, Collection<T>... collec
         return java.util.Collections.unmodifiableList(combined);
     }
 
+    public static <T> List<List<T>> permutations(List<T> items) {
```

Comment:
Permutations tend to be big, so maybe we want to have another method that can lazily compute them. Let's create an issue for it.